TechnicalDocuments/RefactorAppMainUserSession.md

```diff
@@ -0,0 +1,38 @@
+Splitting `AppMainUserSession` into its Logical Components.
+===========================================================
+
+This document describes one possible path for refactoring `AppMainUserSession`. Most likely this refactoring will be carried out in steps by different members of the iOS team. It is therefore important that everyone shares their knowledge and thoughts about how to make `AppMainUserSession` SDK friendly and hopefully more manageable in this document.
+
+`AppMainUserSession` implements `Accessible`, `AuthenticatedAccessible`, `ClinicalAccessible`, `UserSession` and `MainUserSession`. `Accessible` provides basic network calls, `AuthenticatedAccessible` adds network calls authenticated by Kong,  `ClinicalAccessible` are for network calls that require extra authentication to access sensitive data. `UserSession` extends the network access with a `LifeTime` and business controllers for the current user, id verification status and image handling. `MainUserSession`, finally, adds support for accessing family account members, biometric authentication, performing extra authentication to access sensitive information and a signal for observing the current extra authentication status.
+
+Needless to say this a rather complicated entity with poor separation of concerns. `AppMainUserSession` was designed in this way to work around a number of legacy problems, among them multiple authentication domains and that version 2 of the Babylon app relied heavily on singletons. It is, however, doing its job reasonably well and would probably not be marked for refactoring if it was not for the fact that it blocks moving our app away from accessing business controllers directly and instead use the APIs in the Babylon SDK.
+
+More background information about what `MainUserSession` is and what we would like it to become can be found in this [deck](https://drive.google.com/drive/u/0/folders/1PVm5ztVYGpOdqLZxTmIl2BXff8ux8MnK), it is a Keynote presentation so you need download and unzip it.
+
+Proposed refactoring is to change AppMainUserSession from being a singular to a composite entity. It will still comply with the current set of protocols, but implement them by forwarding to the relevant component. This can, hopefully, be done in a step by step manner to avoid a high risk gargantuan pull request.
+
+# Step One, Move to the Babylon Main Project.
+
+Both the SessionViewModel, MainUserSession and AppMainUserSession are specific to the Babylon app targets and not needed to implement the SDK. They should thus be moved out of BabylonCore. To move them to the Octopus folder in the main project seems logical, but BabylonDependencies is also technically possible. Moving them to the main project will require minor refactoring to a couple of UI builders that consume MainUserSession directly.
```

Comment:
We should also move the UserSession protocols to BabylonDeps. They are not used by the SDK at all.

Reply:
It makes sense for the implementation to go to the main app target, btw.